aikit/models/ordinal.py

```diff
@@ -0,0 +1,400 @@
+"""
+
+import numpy as np
+import pandas as pd
+
+from sklearn.base import BaseEstimator, ClassifierMixin, RegressorMixin, is_classifier, is_regressor
+from sklearn.preprocessing import OrdinalEncoder, KBinsDiscretizer
+
+
+from aikit.tools.data_structure_helper import make2dimensions, make1dimension, convert_generic
+from aikit.enums import DataTypes
+from aikit.transformers.categories import _OrdinalOneHotEncoder
+
+
+
+class _BaseOrdinalClassifier(BaseEstimator, ClassifierMixin):
+    """ this class is the base class for Ordinal classifier,
+    it contains methods to convert the target into another format that will be used to fit the underlying classifier
+    
+    """
+    def _prepare_target(self, y, klass, conversion_type):
+        """ prepare the target so that it can be given to the underlying model to use
+        
+        Parameters
+        ----------
+        
+        y : array
+            the original target 
+            
+        klass : type
+            the encoder to use for the target
+            
+        conversion_type : DataType
+            the output type desired by the target
+            
+        Set
+        ---
+        self._mono_target : bool 
+            does the original problem as one target or not
+        self._target_encoded : the encoder used on the target
+        
+        Returns
+        --------
+        y_encoded : array
+            the modified target
+        """
+        self._mono_target = y.ndim == 1
+        self._target_dtype = y.dtype
+        
+        if isinstance(self.classes, str) and self.classes == "auto":
+            categories = "auto"
+        else:
+            if self._mono_target:
+                categories = [self.classes] # because OrdinalEncoder expect a list
+            else:
+                if not isinstance(self.classes, list):
+                    raise TypeError("For multi-target classes should be a list, instead I got %s" % str(type(self.classes)))
+                
+                categories = self.classes
+
+        self._target_encoder = klass(categories=categories, dtype=np.int32)
+        
+        yd2 = convert_generic(make2dimensions(y), output_type=conversion_type)
+  
+        if conversion_type == DataTypes.NumpyArray and yd2.dtype.kind == 'U':
+            yd2 = yd2.astype(np.object, copy=False)
+        
+        y_encoded = self._target_encoder.fit_transform(yd2)
+
+        return y_encoded
+
+
+    @property
+    def classes_(self):
+        if self._mono_target:        
+            return self._target_encoder.categories_[0]
+        else:
+            return self._target_encoder.categories_
+
+class ClassifierFromRegressor(_BaseOrdinalClassifier):
+    """ this class transform a regressor into a classifier
+    it can be used for ordinal classification
+    
+    This model will transform the target into an increasing list of integer
+    and then fit a regression on that
+    """
+
+    def __init__(self,
+                 regressor,
```

Comment:
I wonder if the model shouldn't be called 'model' instead of 'regressor'
+ : regressor make it clear that we are expecting a regressor ....
- :  ... BUT for the BoxCoxTargetTransformer I named it 'model'

what do you think is best ? 'regressor'/'classifier' everywhere or 'model' everywhere ?

Reply:
actually  sklearn uses "estimator" ... so maybe use that everywhere

Reply:
`regressor`/`classifier` is ok for me it's explicit - otherwise use sklearn naming `estimator`